atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -246,14 +246,27 @@ private CassandraClientPoolImpl(
     }
 
     private void tryInitialize() {
-        cassandra.cacheInitialCassandraHosts();
-
+        CassandraVerifierConfig verifierConfig = CassandraVerifierConfig.of(config, runtimeConfig.get());
         // for testability, mock/spy are bad at mockability of things called in constructors
         if (startupChecks == StartupChecks.RUN) {
-            runOneTimeStartupChecks();
+            ensureKeyspaceExists(verifierConfig);
+        }
+
+        ImmutableMap<CassandraServer, CassandraServerOrigin> initialServers =
+                cassandra.getCurrentServerListFromConfig();
+        ImmutableSet<CassandraServer> cassandraServers = initialServers.keySet();
+
+        cassandra.cacheInitialHostsForCalculatingPoolNumber(cassandraServers);
+
+        setServersInPoolTo(initialServers);
```

Comment:
Yeah I don't like it either (it does a lot!), although I do now have a mental association explicitly with setServersInPoolTo so I opted not to change it in this PR.  Don't have a strong preference _against_ changing it though either, since I assume it's the two of us (of code owners) that probably have the strongest association at this point!